include/cpp-statsd-client/StatsdClient.hpp

```diff
@@ -165,39 +170,22 @@ inline void StatsdClient::send(const std::string& key,
         return;
     }
 
-    // Prepare the buffer and include the sampling rate if it's not 1.f
-    const bool needs_dot = !m_prefix.empty() && !key.empty();
-    std::string buffer(256, '\0');
-    int string_len = -1;
-    if (isFrequencyOne) {
-        // Sampling rate is 1.0f, no need to specify it
-        string_len = std::snprintf(&buffer.front(),
-                                   buffer.size(),
-                                   "%s%s%s:%d|%s",
-                                   m_prefix.c_str(),
-                                   needs_dot ? "." : "",
-                                   key.c_str(),
-                                   value,
-                                   type.c_str());
-    } else {
-        // Sampling rate is different from 1.0f, hence specify it
-        string_len = std::snprintf(&buffer.front(),
-                                   buffer.size(),
-                                   "%s%s%s:%d|%s|@%.2f",
-                                   m_prefix.c_str(),
-                                   needs_dot ? "." : "",
-                                   key.c_str(),
-                                   value,
-                                   type.c_str(),
-                                   frequency);
+    // Format the stat message
+    m_buffer.clear();
```

Comment:
Now that we use and alter the buffer, we need to remove the `const` from pretty much all the functions too

Reply:
i think using `mutable` and keeping the methods `const` are a perfectly reasonable here because the caller doesnt percieve any change in state of the client itself. in other words the methods are all still logically const even if they may not be bitwise const. https://en.cppreference.com/w/cpp/language/cv also has a nice little blurb that puts it pretty well:

> Mutable is used to specify that the member does not affect the externally visible state of the class (as often used for mutexes, memo caches, lazy evaluation, and access instrumentation).

Reply:
@vthiery let me know if you'd still rather un`const`ify these methods and remove the `mutable` keyword

Reply:
No you're right sorry for the confusion, `mutable` and keeping `const` is fine.